pyvista/core/pointset.py

```diff
@@ -406,28 +412,116 @@ def is_all_triangles(self):
         --------
         >>> import pyvista
         >>> plane = pyvista.Plane()
-        >>> plane.is_all_triangles()
+        >>> plane.is_all_triangles
         False
 
         >>> sphere = pyvista.Sphere()
-        >>> sphere.is_all_triangles()
+        >>> sphere.is_all_triangles
         True
 
         """
+        class CallableBool(int):  # pragma: no cover
+            """Boolean that can be called.
+
+            Programmer note: We must subclass int and not bool
+            https://stackoverflow.com/questions/2172189/why-i-cant-extend-bool-in-python
+
+            Implemented for backwards compatibility as
+            ``is_all_triangles`` was changed to be a property in
+            ``0.32.0``.
+            """
+
+            def __new__(cls, value):
+                """Use new instead of __init__.
+
+                See:
+                https://jfine-python-classes.readthedocs.io/en/latest/subclass-int.html#emulating-bool-using-new
+
+                """
+                return int.__new__(cls, bool(value))        
+
+            def __call__(self):
+                """Return bool of self and permit calling this bool."""
+                warnings.warn('``is_all_triangles`` is now property as of 0.32.0 and '
+                              'does not need ()', DeprecationWarning)
+                return bool(self)
+
+            def __repr__(self):
+                """Return the string of bool."""
```

Comment:
Shouldn't we make it clear in the `repr` that this is not quite a bool? If someone does end up having a nasty surprise thanks to these shenanigans, at least we should make it easier for them to realise that what they have here isn't actually a bool.

Reply:
These are all good points.  Really, the right answer would be to have not used a property as something callable.  The question is how to change this without messing up workflows, and I'm not able to come up with an elegant solution since python isn't polymorphic (though you can cheat).

The best we can do at this point is to create a class that is nearly like a bool.  I'd change the API outright, but such a depreciation would instantly break any workflows that reply on this method and it wouldn't be clear why as they'd get:

```
TypeError: 'bool' object is not callable
```

I'll clean up this class the best I can, but for this I think:
```
Special cases aren't special enough to break the rules.
Although practicality beats purity.
```